### DISCUSSION TITLE
How use non-learnable parameters / freeze parameters in Flax/JAX?

### DISCUSSION THREAD

**Discussion by marcvanzee** [google/flax#1150] [Q&A] [answered]
I.e. parameters that are part of the model, but not updated after initialization?

(Original question by @dpkingma)

**Comment by marcvanzee** [comment, marked as answer]
The recommended solution is to use a `MultiOptimizer`, our documentation contains an example: https://flax.readthedocs.io/en/latest/flax.optim.html#multioptimizer

Another approach is to use `jax.lax.stop_gradient`: https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.stop_gradient.html

**Reply by andsteing** [reply]
`flax.optim` is deprecated; see #1453 for a solution using Optax instead.
